src/test_suite/multiprocessing_utils.py

```diff
@@ -316,6 +316,8 @@ def build_test_results(
 
             if globals.consensus_mode:
                 harness_ctx.diff_effect_fn = harness_ctx.consensus_diff_effect_fn
+            if globals.core_bpf_mode:
+                harness_ctx.diff_effect_fn = harness_ctx.core_bpf_diff_effect_fn
```

Comment:
This should be fine, since `--consensus` and `--core-bpf` are checked for conflict in the tool's entrypoint.